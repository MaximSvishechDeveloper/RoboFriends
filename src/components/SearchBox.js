import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { search } from "../state/search/searchSlice";

const SearchBox = () => {
  const searchField = useSelector((state) => state.search.value);
  const dispatch = useDispatch();

  const handleSearchChange = (event) => {
    const newValue = event.target.value;
    dispatch(search(newValue));
  };

  return (
    <div className="pa2">
      <input
        value={searchField}
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        onChange={handleSearchChange}
      />
    </div>
  );
};

export default SearchBox;
