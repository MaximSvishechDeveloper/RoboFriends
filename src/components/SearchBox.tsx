import React, { ChangeEvent } from "react";
import { useDispatch, useSelector } from "react-redux";
import { search } from "../state/search/searchSlice";
import { RootState } from "../store";

const SearchBox: React.FC = () => {
  const searchField = useSelector((state: RootState) => state.search.value);
  const dispatch = useDispatch();

  const handleSearchChange = (event: ChangeEvent<HTMLInputElement>) => {
    const newValue = event.target.value;
    dispatch(search(newValue));
  };

  return (
    <div className="pa2">
      <input
        value={searchField}
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        onChange={handleSearchChange}
      />
    </div>
  );
};

export default SearchBox;
