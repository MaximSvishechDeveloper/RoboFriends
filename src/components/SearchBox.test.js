import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";

import SearchBox from "./SearchBox";
import searchReducer, { search } from "../state/search/searchSlice";

describe("SearchBox Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = configureStore({
      reducer: {
        search: searchReducer,
      },
    });

    component = mount(
      <Provider store={store}>
        <SearchBox />
      </Provider>
    );
  });

  it("renders the search box correctly", () => {
    const searchInput = component.find('input[type="search"]');
    expect(searchInput).toHaveLength(1);
  });

  it("dispatches the search action on input change", () => {
    const searchInput = component.find('input[type="search"]');
    searchInput.simulate("change", { target: { value: "test" } });

    const dispatchedActions = store.getState().search;

    expect(dispatchedActions).toEqual({ value: "test" });
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
