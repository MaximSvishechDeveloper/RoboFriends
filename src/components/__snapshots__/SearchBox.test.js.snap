// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchBox>
    <div
      className="pa2"
    >
      <input
        className="pa3 ba b--green bg-lightest-blue"
        onChange={[Function]}
        placeholder="search robots"
        type="search"
        value=""
      />
    </div>
  </SearchBox>
</Provider>
`;
